### ISSUE TITLE
Can "cargo script" be used to run examples from existing projects that require additional heavy dependencies?

### ISSUE THREAD

**Issue by vi** [DanielKeep/cargo-script#12]
Imagine lightweight project where one of example demonstrate it's use with a heavyweight thing.

You don't want to point this dependency even in dev-deps, but still want the example to be easily runnable.

Can `cargo script` help in this case?

**Comment by DanielKeep** [comment]
Well yes, but `cargo script` isn't going to treat the example any differently to any other random Rust source file.  That it's part of a package is irrelevant; the example would need to list the package itself as an explicit dependency.

Is there anything specific that you're suggesting?

**Comment by vi** [comment]
Can one list cargo-script as a dev-depencency of a project and make `cargo run --example myexample` deviate to cargo-script instead of trying to run example with missing dependencies directly?

**Comment by DanielKeep** [comment]
No; there's no facility in cargo for anything like that at all.  `cargo script` isn't intended to be used on packages *at all*; it's whole remit is to run isolated source files.

I'm going to close this, because I *think* what you actually want is a way to attach dependencies to specific examples.  This would need to be a feature of Cargo *itself*, rather than `cargo script`.

**Comment by vi** [comment]
When I was suggesting per-example dependencies on IRC, they pointed me at cargo-script, that's why I was asking.

**Comment by DanielKeep** [comment]
It's just that "examples" are are a concept directly supported by Cargo.  `cargo script` can't really do anything useful with them, because it's not involved in building or running them.

Random, loose example programs that aren't "examples" in the official sense would work just like any other random, loose script that `cargo script` can run, so there shouldn't be anything that needs to be added for those.

In other words, additional features in this area are something that needs to be added to *Cargo*, not `cargo script` (which is just an external tool that happens to be runnable via Cargo).
